core/domain/exp_domain.py

```diff
@@ -1455,6 +1432,8 @@ def update_interaction_answer_groups(self, answer_groups_list):
                     rule_inputs[param_name] = normalized_param
 
                 answer_group.rule_specs.append(rule_spec)
+            if answer_group_dict['training_data']:
```

Comment:
Why do you need the "if" check?

Reply:
Older explorations don't have this attribute and so I have this if check because we are generating directly from a dict and it might belong to an older schema version where training data attribute does not exist.

Reply:
Oh, no, all these domain methods assume that an exploration with the latest schema version is being acted on. When an exploration is converted to a domain object it is migrated to the latest schema version first. So you can assume that the exploration object uses the most recent schema version.

Reply:
Yes, but this is a dict not an exploration domain object and I don't know where that update method is actually used but test cases fail if I don't do this check. I don't know where that answer group list is supposed to come from.

Reply:
So I checked the test cases and mistakes are with the new answer groups defined in test cases. I will remove this condition in next commit.

Reply:
OK, thanks.

Reply:
Done